examples/asr/asr_adapters/train_asr_adapter.py

```diff
@@ -142,8 +142,15 @@ def main(cfg):
 
     # Setup optimizer
     cfg.model.optim = update_model_cfg(model.cfg.optim, cfg.model.optim)
+    model.cfg.optim = cfg.model.optim  # Update the model's config
     model.setup_optimization(cfg.model.optim)
```

Comment:
Thats what the update_model_cfg() does - it looks up the old config inside the model, and merges the config with the values inside the user defined overrides. It wont allow deletion or addition but only override.

If we want to support addition and deletion then we have cannot use the internal config at all. We are only left with the user having to explicitly define everything or nothing.

Reply:
What I am saying is that by default I think defining those fully in the config makes more sense, as it would allow completely different classes for those, which would be useful. For a specific example, the default models likely use noam, which is not ideal if we're only doing a few adapter training steps; but for other optimizers we'd need to have different parameters, such as removing d_model from optimizer config. If we do still really want to keep the config merge functionality, perhaps we can have a special flag for overriding configs.

Reply:
In fact, thinking also about potentially applying adapters to ssl models, it would be even more important there to be able to completely re-do these configs, as things such as masking etc. may be completely different

Reply:
Hmm valid points. Ok we will go the "complete replacement approach" then.

Reply:
Done